### ISSUE TITLE
Invalid $OLDPWD: command "cd -" not working anymore

### ISSUE THREAD

**Issue by poncki** [kennethreitz/autoenv#141]
Hello.

While fixing #125 an error was introduced in ae84a558eceb0ec39db7cd43c82e15cc81176e44.
After `cd <dir>` the $OLDPWD variable is set to a wrong value (a new directory, not the old one) and `cd -` is not working anymore. _Probably_ this is an effect of setting $PWD in line 14...

pnc

**Comment by dasJ** [comment]
Does #145 fix this for you?

**Comment by carhartl** [comment]
I'm having the same issue, but I'm not sure whether that may be my zsh setup or not...

**Comment by carhartl** [comment]
All I could find out so far is that when sourcing activate.sh `$OLDPWD == $PWD`, happening when `autoenv_init` is executed... for testing I pasted the latest code from master...

**Comment by carhartl** [comment]
So basically what I did to fix this was moving this line: https://github.com/kennethreitz/autoenv/blob/7eb70eca0039f757525a2f88f008e647a492d7f1/activate.sh#L14

_into_ the subshell:

```
  # Discover all files we need to source
  # We do this in a subshell so we can cd/chdir
  _files="`
    command -v chdir >/dev/null 2>&1 && \chdir "${_pwd}" || builtin cd "${_pwd}"
    _hadone=''
    while :; do
      _file="$(pwd -P)/${AUTOENV_ENV_FILENAME}"
      if [ -f "${_file}" ]; then
        if [ -z "${_hadone}" ]; then
          echo -n "${_file}"
          _hadone='1'
        else
          echo -n "
${_file}"
        fi
      fi
      [ "$(pwd -P)" = "${_mountpoint}" ] && break
      command -v chdir >/dev/null 2>&1 && \chdir "$(pwd -P)/.." || builtin cd "$(pwd -P)/.."
    done
```

Afterwards the value of `$OLDPWD` was intact and `cd -` worked again. Everything seems to work, though I didn't run any tests yet, just tried it out with an existing .env file.

**Comment by n8henrie** [comment]
Also suddenly noticed `cd -` not working, eventually discovered `$OLDPWD` and can confirm it was incorrect.

Took a fair amount of Googling to find this, the patch fixes the issue for me as well.

**Comment by dasJ** [comment]
As #149 is merged, is this fixed for you?

**Comment by carhartl** [comment]
When homebrew user,while waiting for a new release to be pushed, something like this formalu may help: https://gist.github.com/carhartl/495c498aa1a1ef8a561d2ad53c52b3d6

You'd need to execute `brew instal --devel autoenv`

**Comment by carhartl** [comment]
Though yeah, might be easier to source activate.sh from somewhere else!

**Comment by carhartl** [comment]
And the easiest seems to be:

    $ brew uninstall autoenv
    $ brew install --HEAD autoenv
    $ brew link autoenv

**Comment by n8henrie** [comment]
Yes, issue now seems resolved by installing from `HEAD`. Thank you!

**Comment by henrahmagix** [comment]
I just came across this problem myself, and installing from HEAD has worked.

Can there please be a stable release soon to homebrew?

**Comment by pv8** [comment]
The release [v0.2.1](https://github.com/kennethreitz/autoenv/releases/tag/v0.2.1) doesn't actually include the fix for issue #141. 

This release is on commit 92d2f3b (Oct 18, 2016) and the fix is on 6c55ec3 (Nov 11, 2016).

**Comment by dyson** [comment]
This just go me as well. Would be really great to get a new release out.
